version: "3.3"
services:
    traefik:
        image: traefik:v2.4
        restart: unless-stopped
        container_name: traefik
        ports:
            - "80:80"
            - "443:443"


        command:
            ## API Settings ##
            - --api.insecure=true
            - --api.dashboard=true
            - --api.debug=true
            - --log.level=ERROR    
            ##Provider Settings##
            - --providers.docker=true
            - --providers.docker.exposedbydefault=false
            - --providers.file.filename=/dynamic.yaml
            - --providers.docker.network=proxy
            ## Entrypoints ##
            - --entrypoints.http.address=:80
            - --entrypoints.https.address=:443
            #- --entrypoints.mc.address=:25565   #uncomment if you have Minecraft Servers behind traefik

            ##Certificate settings##
            - --certificatesresolvers.mytlschallenge.acme.tlschallenge=true
            - --certificatesresolvers.mytlschallenge.acme.email=email@domain.de  # add your own mail here
            - --certificatesresolvers.mytlschallenge.acme.storage=/letsencrypt/acme.json

        volumes:
            - ./letsencrypt:/letsencrypt
            - /var/run/docker.sock:/var/run/docker.sock
            - ./dynamic.yaml:/dynamic.yaml
        networks:
            - proxy

        labels:
            - "traefik.enable=true"
            - "traefik.http.routers.api.rule=Host(`traefik.yourdomain.com`)"   ## add your domain for the traefik dashboard
            - "traefik.http.routers.traefik.entrypoints=http"
            - "traefik.http.routers.api.service=api@internal"


networks:
    proxy:
        external: true
